cmd/common/registry.go

```diff
@@ -238,7 +238,9 @@ func buildCommandFromGadget(
 
 			formatter := parser.GetTextColumnsFormatter()
 			if outputModeParams != "" {
-				formatter.SetShowColumns(strings.Split(outputModeParams, ","))
+				if err := formatter.SetShowColumns(strings.Split(outputModeParams, ",")); err != nil {
```

Comment:
Should we really error out in this case, though? Maybe a warning would suffice? If for example I'm switching between gadgets but defined a list of all columns I'm interested in, this would punish and force me to "clean up" that list. We have a VerifyColumns() that could be used to soft-fail (only forwarding the valid columns and reporting the invalid ones). This was btw also the intention behind the soft-fail inside SetShowColumns() in the first place.

However, I like your version (returning an error) more because it's cleaner - and we don't lose the soft-fail as said when combining with the verifier function.

Reply:
Makes sense to me. I updated the PR.